clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -307,9 +307,8 @@ public void updateFetchPositions(Set<TopicPartition> partitions) {
                         if (error == Errors.INVALID_TOPIC_EXCEPTION)
                             throw new InvalidTopicException("Topic '" + topic + "' is invalid");
                         else if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION)
-                            // if a requested topic is unknown, we just continue and let it be absent
-                            // in the returned map
-                            continue;
+                            throw new KafkaException("Topic '" + topic + "' may not exist or the user may not have " +
```

Comment:
Sorry for the confusion. I will edit the PR to account for this error from offset fetch API only. And whether to throw an exception in that situation or not is also up for discussion as @hachikuji pointed out in the corresponding JIRA.